File: backend/schemas/triage.py
"""
Triage Schemas

分诊相关的 Pydantic 模型。
"""
from datetime import datetime
from typing import Optional, List
from enum import Enum
from pydantic import BaseModel, Field


class DifficultyLevelEnum(str, Enum):
    """难度级别枚举"""
    L1 = "L1"  # 入门级
    L2 = "L2"  # 基础级
    L3 = "L3"  # 中级
    L4 = "L4"  # 高级
    L5 = "L5"  # 专家级


class TriageDecision(str, Enum):
    """分诊决策枚举"""
    AUTO_REPLY = "auto_reply"           # 自动回复
    AUTO_REPLY_CONFIRM = "auto_reply_confirm"  # 自动回复需确认
    TO_ASSISTANT = "to_assistant"       # 转助教
    TO_TEACHER = "to_teacher"           # 转教师
    TO_TEACHER_URGENT = "to_teacher_urgent"  # 紧急转教师


class HandlerRole(str, Enum):
    """处理人角色枚举"""
    SYSTEM = "system"       # 系统自动
    ASSISTANT = "assistant" # 助教
    TEACHER = "teacher"     # 教师


# 难度级别描述
DIFFICULTY_DESCRIPTIONS = {
    "L1": {"name": "入门级", "description": "基础语法、简单概念", "examples": ["变量声明", "数据类型", "print语句"]},
    "L2": {"name": "基础级", "description": "常用语法、基本操作", "examples": ["循环", "条件判断", "函数定义"]},
    "L3": {"name": "中级", "description": "进阶概念、常见算法", "examples": ["递归", "排序算法", "数据结构"]},
    "L4": {"name": "高级", "description": "复杂问题、设计模式", "examples": ["并发编程", "架构设计", "性能优化"]},
    "L5": {"name": "专家级", "description": "系统级问题、深度优化", "examples": ["分布式系统", "底层原理", "高级调优"]},
}


class TriageRequest(BaseModel):
    """分诊请求"""
    question: str = Field(..., min_length=1, description="用户问题")
    user_id: Optional[str] = None
    user_name: Optional[str] = None
    session_id: Optional[str] = None
    is_urgent: bool = Field(False, description="是否紧急")
    context: Optional[str] = Field(None, description="上下文信息")


class TriageResponse(BaseModel):
    """分诊响应"""
    log_id: str
    question: str
    detected_category: Optional[str] = None
    detected_difficulty: int
    difficulty_label: str
    match_score: float
    matched_entry_id: Optional[str] = None
    decision: TriageDecision
    answer: Optional[str] = None
    answer_source: Optional[str] = None
    assigned_to: Optional[str] = None
    priority: int
    is_urgent: bool
    confidence_message: str
    created_at: datetime


class PendingQuestion(BaseModel):
    """待处理问题"""
    log_id: str
    question: str
    user_id: Optional[str] = None
    user_name: Optional[str] = None
    detected_category: Optional[str] = None
    detected_difficulty: int
    difficulty_label: str
    match_score: Optional[float] = None
    priority: int
    is_urgent: bool
    triage_result: str
    created_at: datetime
    waiting_time_seconds: float


class PendingQueueResponse(BaseModel):
    """待处理队列响应"""
    total: int
    urgent_count: int
    questions: List[PendingQuestion]


class TeacherTakeoverRequest(BaseModel):
    """教师接管请求"""
    log_id: str
    teacher_id: str
    teacher_name: Optional[str] = None


class TeacherAnswerRequest(BaseModel):
    """教师回答请求"""
    log_id: str
    teacher_id: str
    answer: str = Field(..., min_length=1)
    update_knowledge_base: bool = Field(False, description="是否更新知识库")
    new_keywords: Optional[List[str]] = None


class TeacherAnswerResponse(BaseModel):
    """教师回答响应"""
    log_id: str
    answer: str
    answered_by: str
    answered_at: datetime
    knowledge_base_updated: bool
    new_entry_id: Optional[str] = None


class TriageStats(BaseModel):
    """分诊统计"""
    total_questions: int
    auto_replied: int
    to_assistant: int
    to_teacher: int
    pending: int
    urgent_pending: int
    avg_response_time: Optional[float] = None
    avg_waiting_time: Optional[float] = None
    resolution_rate: float
    questions_by_difficulty: dict
    questions_by_decision: dict


class DifficultyInfo(BaseModel):
    """难度级别信息"""
    level: str
    name: str
    description: str
    examples: List[str]


class DifficultyLevelsResponse(BaseModel):
    """难度级别列表响应"""
    levels: List[DifficultyInfo]

File: frontend-react-backup/src/types/triage.ts
/**
 * 分诊相关类型定义
 */

// 分诊决策枚举
export type TriageDecision =
  | 'auto_reply'
  | 'auto_reply_confirm'
  | 'to_assistant'
  | 'to_teacher'
  | 'to_teacher_urgent';

// 问答日志状态枚举
export type QALogStatus = 'answered' | 'pending' | 'escalated' | 'closed';

// 分诊结果枚举
export type TriageResult = 'auto_reply' | 'to_assistant' | 'to_teacher' | 'pending';

// 分诊决策标签映射
export const TRIAGE_DECISION_LABELS: Record<TriageDecision, string> = {
  auto_reply: '自动回复',
  auto_reply_confirm: '自动回复（待确认）',
  to_assistant: '转助教',
  to_teacher: '转教师',
  to_teacher_urgent: '紧急转教师',
};

// 状态标签映射
export const STATUS_LABELS: Record<QALogStatus, string> = {
  answered: '已回答',
  pending: '待处理',
  escalated: '已升级',
  closed: '已关闭',
};

// 分诊请求
export interface TriageRequest {
  question: string;
  user_id?: string;
  user_name?: string;
  session_id?: string;
  is_urgent?: boolean;
  context?: string;
}

// 分诊响应
export interface TriageResponse {
  log_id: string;
  question: string;
  detected_category: string | null;
  detected_difficulty: number;
  difficulty_label: string;
  match_score: number;
  matched_entry_id: string | null;
  decision: TriageDecision;
  answer: string | null;
  answer_source: string | null;
  assigned_to: string | null;
  priority: number;
  is_urgent: boolean;
  confidence_message: string;
  created_at: string;
}

// 待处理问题
export interface PendingQuestion {
  log_id: string;
  question: string;
  user_id: string | null;
  user_name: string | null;
  detected_category: string | null;
  detected_difficulty: number;
  difficulty_label: string;
  match_score: number | null;
  priority: number;
  is_urgent: boolean;
  triage_result: string;
  created_at: string;
  waiting_time_seconds: number;
}

// 待处理队列响应
export interface PendingQueueResponse {
  total: number;
  urgent_count: number;
  questions: PendingQuestion[];
}

// 教师接管请求
export interface TeacherTakeoverRequest {
  log_id: string;
  teacher_id: string;
  teacher_name?: string;
}

// 教师回答请求
export interface TeacherAnswerRequest {
  log_id: string;
  teacher_id: string;
  answer: string;
  update_knowledge_base?: boolean;
  new_keywords?: string[];
}

// 教师回答响应
export interface TeacherAnswerResponse {
  log_id: string;
  answer: string;
  answered_by: string;
  answered_at: string;
  knowledge_base_updated: boolean;
  new_entry_id: string | null;
}

// 分诊统计
export interface TriageStats {
  total_questions: number;
  auto_replied: number;
  to_assistant: number;
  to_teacher: number;
  pending: number;
  urgent_pending: number;
  avg_response_time: number | null;
  avg_waiting_time: number | null;
  resolution_rate: number;
  questions_by_difficulty: Record<string, number>;
  questions_by_decision: Record<string, number>;
}

// 难度级别信息
export interface DifficultyInfo {
  level: string;
  name: string;
  description: string;
  examples: string[];
}

// 难度级别列表响应
export interface DifficultyLevelsResponse {
  levels: DifficultyInfo[];
}

